functions/output_lib.sh

```diff
@@ -0,0 +1,190 @@
+}
+
+logjson (){
+  printf "\n  \"%s\": \"%s\"," "$1" "$2" | tee -a "$logger.json" 2>/dev/null 1>&2
+}
+
+SSEP=
+SEP=
+startsectionjson() {
+  printf "%s\n    {\n      \"id\": \"%s\",\n      \"desc\": \"%s\",\n      \"results\": [" "$SSEP" "$1" "$2" | tee -a "$logger.json" 2>/dev/null 1>&2
+  SEP=
+  SSEP=","
+}
+
+endsectionjson() {
+  printf "\n      ]\n    }" | tee -a "$logger.json" 2>/dev/null 1>&2
+}
+
+starttestjson() {
+  printf "%s\n        {\n          \"id\": \"%s\",\n          \"desc\": \"%s\",\n          " "$SEP" "$1" "$2" | tee -a "$logger.json" 2>/dev/null 1>&2
+  SEP=","
+}
+
+log_to_json() {
+  if [ $# -eq 1 ]; then
+    printf "\"result\": \"%s\"" "$1" | tee -a "$logger.json" 2>/dev/null 1>&2
+    return 
+  fi
+  if [ $# -eq 2 ] && [ $# -ne 1 ]; then
+    # Result also contains details
+    printf "\"result\": \"%s\",\n          \"details\": \"%s\"" "$1" "$2" | tee -a "$logger.json" 2>/dev/null 1>&2
+    return
+  fi
+  # Result also includes details and a list of items. Add that directly to details and to an array property "items"
+  # Also limit the number of items to $limit, if $limit is non-zero
+  truncItems=$3
+  if [ $limit != 0 ]; then
+    truncItems=""
+    ITEM_COUNT=0
+    for item in $3; do
+      truncItems="$truncItems $item"
+      ITEM_COUNT=$((ITEM_COUNT + 1));
+      if [ "$ITEM_COUNT" == "$limit" ]; then
+        truncItems="$truncItems (truncated)"
+        break;
+      fi
+    done
+  fi
+  itemsJson=$(printf "[\n            "; ISEP=""; ITEMCOUNT=0; for item in $truncItems; do printf "%s\"%s\"" "$ISEP" "$item"; ISEP=","; done; printf "\n          ]")
+  printf "\"result\": \"%s\",\n          \"details\": \"%s: %s\",\n          \"items\": %s" "$1" "$2" "$truncItems" "$itemsJson" | tee -a "$logger.json" 2>/dev/null 1>&2
+}
+
+logcheckresult() {
+  # Log to JSON
+  log_to_json "$@"
+
+  # Log remediation measure to JSON
+  if [ -n "$remediation" ] && [ "$1" != "PASS" ] && [ "$printremediation" = "1" ]; then
+    printf ",\n          \"remediation\": \"%s\"" "$remediation" | tee -a "$logger.json" 2>/dev/null 1>&2
+    if [ -n "$remediationImpact" ]; then
+      printf ",\n          \"remediation-impact\": \"%s\"" "$remediationImpact" | tee -a "$logger.json" 2>/dev/null 1>&2
+    fi
+  fi
+  printf "\n        }" | tee -a "$logger.json" 2>/dev/null 1>&2
+
+  # Save remediation measure for print log to stdout
+  if [ -n "$remediation" ] && [ "$1" != "PASS" ]; then
+    if [ -n "${checkHeader}" ]; then
+      if [ -n "${addSpaceHeader}" ]; then
+        globalRemediation="${globalRemediation}\n"
+      fi
+      globalRemediation="${globalRemediation}\n${bldblu}[INFO]${txtrst} ${checkHeader}"
+      checkHeader=""
+      addSpaceHeader="1"
+    fi
+    globalRemediation="${globalRemediation}\n${bldblu}[INFO]${txtrst} ${id} - ${remediation}"
+    if [ -n "${remediationImpact}" ]; then
+      globalRemediation="${globalRemediation} Impact: ${remediationImpact}"
```

Comment:
I meant to change this `Impact` to `Remediation Impact` as in the json file.

Reply:
@konstruktoid Thanks for the feedback. I made this change.